objcgen/objcgenerator.cs

```diff
@@ -562,6 +562,18 @@ void GenerateArgumentArray (string parameterName, string argumentName, Type t, r
 				implementation.WriteLine ("else");
 				implementation.Indent++;
 				implementation.WriteLine ($"mono_array_set ({pnameArr}, {typeName}, {pnameIdx}, {pnameRet}.{returnValue});");
+				implementation.Indent--;
+				break;
+			case TypeCode.Decimal:
+				implementation.WriteLine ($"NSDecimalNumber* {pnameRet} = {parameterName}[{pnameIdx}];");
+				implementation.WriteLine ($"if (!{pnameRet} || [{pnameRet} isKindOfClass:[NSNull class]])");
+				implementation.Indent++;
+				implementation.WriteLine ($"continue;");
+				implementation.Indent--;
+				implementation.WriteLine ("else");
```

Comment:
do you need the `else` after the `continue` ?
if not then we can reduce the code (and indentation level)